### PR TITLE
Initial support for OCI multi-platform image

### PR DESCRIPTION
**Pull request by arm64b**
Add the OCI spec compatible image support in client side.

Signed-off-by: Dennis Chen <dennis.chen@arm.com>

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by arm64b** [comment]
Will check the ci-failing tomorrow

**Comment by thaJeztah** [comment]
Thanks!

File: api/server/router/build/build_routes.go [hunk 1]
```diff
@@ -72,11 +72,13 @@ func newImageBuildOptions(ctx context.Context, r *http.Request) (*types.ImageBui
 	options.RemoteContext = r.FormValue("remote")
 	if versions.GreaterThanOrEqualTo(version, "1.32") {
 		apiPlatform := r.FormValue("platform")
-		p := system.ParsePlatform(apiPlatform)
-		if err := system.ValidatePlatform(p); err != nil {
-			return nil, errdefs.InvalidParameter(errors.Errorf("invalid platform: %s", err))
+		if len(strings.TrimSpace(apiPlatform)) != 0 {
```
Comment (kolyshkin) [inline api/server/router/build/build_routes.go]: Where is this (`strings.TrimSpace()`) coming from? It seems you are relaxing the spec here, allowing for non-empty strings containing some whitespace characters. Previous implementation (`system.ParsePlatform()`) did not do any special treatment of white space.

File: api/server/router/image/image_routes.go [hunk 2]
```diff
@@ -45,9 +44,12 @@ func (s *imageRouter) postImagesCreate(ctx context.Context, w http.ResponseWrite
 	version := httputils.VersionFromContext(ctx)
 	if versions.GreaterThanOrEqualTo(version, "1.32") {
 		apiPlatform := r.FormValue("platform")
-		platform = system.ParsePlatform(apiPlatform)
-		if err = system.ValidatePlatform(platform); err != nil {
-			err = fmt.Errorf("invalid platform: %s", err)
+		if len(strings.TrimSpace(apiPlatform)) != 0 {
```
Comment (kolyshkin) [inline api/server/router/image/image_routes.go]: ditto

File: image/tarexport/load.go [hunk 3]
```diff
@@ -422,8 +422,6 @@ func checkCompatibleOS(imageOS string) error {
 		return fmt.Errorf("cannot load %s image on %s", imageOS, runtime.GOOS)
 	}
 	// Finally, check the image OS is supported for the platform.
-	if err := system.ValidatePlatform(system.ParsePlatform(imageOS)); err != nil {
-		return fmt.Errorf("cannot load %s image on %s: %s", imageOS, runtime.GOOS, err)
-	}
+	// TODO(@arm64b): Leave this sanity check to the containerd code in the future
```
Comment (kolyshkin) [inline image/tarexport/load.go]: From the code it appears it's not longer a TODO item, so maybe just remove the comments about checking altogether?

**Review by kolyshkin — CHANGES_REQUESTED** [review]

**Comment by kolyshkin** [comment]
CI failures are caused by
```
02:06:53 # github.com/docker/docker/client
02:06:53 client/image_build.go:33:22: invalid operation: options.Platform != "" (mismatched types v1.Platform and string)
02:06:53 client/image_build.go:37:32: cannot use options.Platform (type v1.Platform) as type string in argument to query.Set
02:06:53 client/image_build.go:133:22: invalid operation: options.Platform != "" (mismatched types v1.Platform and string)
02:06:53 client/image_build.go:134:48: cannot use options.Platform (type v1.Platform) as type string in argument to strings.ToLower
```

**Comment by arm64b** [comment]
Hello @kolyshkin , thanks for the comments! I think we can close this PR since we have a replacement of PR #37350 by @tonistiigi 👍

**Pull request closed without merging**